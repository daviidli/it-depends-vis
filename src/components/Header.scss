.headerContainer {
	margin: 12px;

	#repoSend {
		height: 56px;
		margin-left: 4px;
	}
}

