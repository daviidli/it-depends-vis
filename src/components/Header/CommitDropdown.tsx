import React from 'react';
import Dropdown from '../Dropdown/Dropdown';
import { ICommit } from '../../App';

export interface CommitDropdownProps {
	label: string,
	selectedCommit: string,
	showCommitsAfter: string,
	setSelectedCommit: Function,
	commits: ICommit[],
};

const CommitDropdown = (props: CommitDropdownProps) => {
	const { label, selectedCommit, showCommitsAfter, setSelectedCommit, commits } = props;

	let commitsShown: ICommit[];
	const indexOfStart: number = indexOfCommit(commits, showCommitsAfter);
	if (indexOfStart < 0) {
		commitsShown = commits;
	} else {
		commitsShown = commits.slice(indexOfStart + 1, commits.length);
	}

	const commitsShownValues = commitsShown.map((commit: ICommit) => commit.value);

	return (
		<Dropdown
			selectedValue={selectedCommit}
			label={label}
			disabled={!commitsShown.length}
			setSelectedValue={setSelectedCommit}
			values={commitsShownValues}
		/>
	);
}

export const indexOfCommit = (commits: ICommit[], commitName: string) => {
	for (let i = 0; i < commits.length; i++) {
		if (commits[i].value === commitName) {
			return i;
		}
	}
	return -1;
};

export default CommitDropdown;
