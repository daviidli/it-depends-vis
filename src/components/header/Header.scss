.header {
	background-color: #fff;
	box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.06), 0 6px 20px 0 rgba(0, 0, 0, 0.02);

	.icon {
		margin-top: 3px;
	}

	.ant-page-header-heading-left {
		width: 30em;
	}

	.ant-page-header-heading-extra {
		width: 90%;
	}

	.contents {
		width: 100%;

		.dropdownContainer {
			display: flex;
			justify-content: center;
		}
	}
}
