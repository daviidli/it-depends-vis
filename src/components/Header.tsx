import React, { useState } from 'react';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

import './Header.scss';

const Header: React.FC = () => {
	const [repo, setRepo] = useState('');

	const handleKeyDown = (e: any) => {
		if (e.key === 'Enter') {
			console.log(repo);
			return;
		}
	};

	return (
		<div className='headerContainer'>
			<Grid
				container
				className='repoContainer'
				spacing={0}
				justify='flex-start'
			>
				<Grid item xs={10} sm={11} md={4}>
					<TextField
						label='Repository'
						type='search'
						variant='outlined'
						fullWidth={true}
						value={repo}
						onChange={(e) => setRepo(e.target.value)}
						onKeyDown={handleKeyDown}
					/>
				</Grid>
				<Grid item xs={2} sm={1} md={1}>
					<Button
						id='repoSend'
						variant="contained"
						color="secondary"
					>
						>
					</Button>
				</Grid>
			</Grid>
		</div>
		
	);
}

export default Header;