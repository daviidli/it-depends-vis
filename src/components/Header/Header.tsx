import React, { useState, useContext } from 'react';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import HelpOutlineIcon from '@material-ui/icons/HelpOutline';
import Dropdown from '../Dropdown/Dropdown';
import { HelpModalContext } from '../Modal/HelpModalContext';

import './Header.scss';

const Header = () => {
	const [repo, setRepo] = useState('');
	const [commits, setCommits] = useState([] as string[]);
	const [selectedCommits, setSelectedCommits] = useState({ start: '', end: '' });
	const setOpen: Function = useContext(HelpModalContext)[1];

	const sendRequest = () => {
		// TODO
		console.log(repo);
		setCommits([
			'commit #1',
			'commit #2',
			'commit #3',
			'commit #4',
			'commit #5',
			'commit #6',
			'commit #7',
			'commit #8',
			'commit #9',
			'commit #10'
		]);
	};

	const handleKeyDown = (e: any) => {
		if (e.key === 'Enter') {
			sendRequest();
		}
	};

	const setStartCommit = (commit: string) => {
		setSelectedCommits(prev => ({ ...prev, start: commit }));
	};

	const setEndCommit = (commit: string) => {
		setSelectedCommits(prev => ({ ...prev, end: commit }));
	};

	const showHelp = () => {
		setOpen(true);
	};

	return (
		<div className='headerContainer'>
			<Grid
				container
				className='repoContainer'
				spacing={1}
				justify='space-evenly'
			>
				<Grid item xs={12}>
					<span className='heading'>IT DEPENDS</span>
					<IconButton className='help' aria-label='help' onClick={showHelp}>
						<HelpOutlineIcon />
					</IconButton>
				</Grid>
				<Grid item xs={10} sm={11} md={4}>
					<TextField
						label='Repository'
						type='search'
						variant='outlined'
						margin='dense'
						fullWidth={true}
						value={repo}
						onChange={(e) => setRepo(e.target.value)}
						onKeyDown={handleKeyDown}
					/>
				</Grid>
				<Grid item xs={2} sm={1} md={1}>
					<Button
						id='repoSend'
						variant='contained'
						color='secondary'
						onClick={() => sendRequest()}
					>
						>
					</Button>
				</Grid>
				<Grid item xs={12} sm={12} md={3}></Grid>
				<Grid item xs={6} sm={6} md={2}>
					<Dropdown
						label='Start Commit'
						selectedCommit={selectedCommits.start}
						showCommitsAfter={''}
						setSelectedCommit={setStartCommit}
						commits={commits}
					/>
				</Grid>
				<Grid item xs={6} sm={6} md={2}>
					<Dropdown
						label='End Commit'
						selectedCommit={selectedCommits.end}
						showCommitsAfter={selectedCommits.start}
						setSelectedCommit={setEndCommit}
						commits={commits}
					/>
				</Grid>
			</Grid>
		</div>
		
	);
}

export default Header;