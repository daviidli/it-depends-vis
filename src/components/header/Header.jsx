import React from 'react';
import {
	PageHeader, Button, Row, Col
} from 'antd';
import { GiThink } from 'react-icons/gi';
import { FiGitCommit } from 'react-icons/fi';
import { useHistory } from 'react-router';
import routes from '../../constants/routes.json';
import SearchBox from '../../containers/SearchBoxContainer';
import Slider from '../../containers/SliderContainer';
import Dropdown from '../../containers/DropdownContainer';
import './Header.scss';

const Header = () => {
	const history = useHistory();

	const onBack = () => history.push(routes.HOME);
	const backIcon = <Button className="icon" type="primary" shape="circle" size="large" icon={<GiThink size={32} />} />;

	const contents = (
		<Row className="contents" justify="end">
			<Col />
			<Col xs={22} md={12}><Slider /></Col>
			<Col xs={2} className="dropdownContainer">
				<Dropdown icon={<FiGitCommit size={28} />} />
			</Col>
		</Row>
	);

	return (
		<PageHeader
			className="header"
			title={<SearchBox />}
			backIcon={backIcon}
			onBack={onBack}
			extra={contents}
		/>
	);
};

export default Header;
