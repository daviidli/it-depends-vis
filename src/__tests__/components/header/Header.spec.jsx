import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { useHistory } from 'react-router';
import { PageHeader } from 'antd';
import Header from '../../../components/header/Header';

jest.mock('react-router');

const setup = () => {
	const wrapper = shallow(<Header />);
	return {
		wrapper,
		button: wrapper.find(PageHeader)
	};
};

describe('Header component', () => {
	it('should match snapshot', () => {
		useHistory.mockImplementation(jest.fn());
		const { wrapper } = setup();
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should go home on button click', () => {
		const mockPush = jest.fn();
		useHistory.mockImplementation(() => ({ push: mockPush }));
		const { button } = setup();
		expect(mockPush).toHaveBeenCalledTimes(0);
		button.simulate('back');
		expect(mockPush).toHaveBeenCalledTimes(1);
		expect(mockPush).toHaveBeenCalledWith('/');
	});
});
