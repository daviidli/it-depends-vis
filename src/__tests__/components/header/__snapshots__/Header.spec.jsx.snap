// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<PageHeader
  backIcon={
    <Button
      block={false}
      className="icon"
      ghost={false}
      htmlType="button"
      icon={
        <GiThink
          size={32}
        />
      }
      loading={false}
      shape="circle"
      size="large"
      type="primary"
    />
  }
  className="header"
  extra={
    <Row
      className="contents"
      gutter={0}
      justify="end"
    >
      <Col />
      <Col
        md={12}
        xs={22}
      >
        <UNDEFINED />
      </Col>
      <Col
        className="dropdownContainer"
        xs={2}
      >
        <UNDEFINED
          icon={
            <FiGitCommit
              size={28}
            />
          }
        />
      </Col>
    </Row>
  }
  onBack={[Function]}
  title={<UNDEFINED />}
/>
`;
